pdf/pdfdocument.cpp

```diff
@@ -315,9 +318,17 @@ void PDFDocument::jobFinished(PDFJob *job)
 
 void PDFDocument::searchFinished(const QList<QPair<int, QRectF>> &matches)
 {
-    delete d->searchModel;
-    d->searchModel = new PDFSearchModel(matches);
-    emit searchModelChanged();
+    if (matches.count() != d->searchModel->count())
```

Comment:
Alright, I've removed from the searchFinished signal the reference to the complete list, since not used anymore. It will be easy to readd it later if necessary.

> if there thread didn't store any matches itself.

Well, currently the search thread stores the full list, while building it and I keep it like that (as you said for index handling simplicity). But after searchFinished has been emitted, the list in the thread is not needed anymore, so may be cleared. Nevertheless, since it's the main thread that copies the list items from the search thread list, I'm afraid the searchFinished may be reached in the search thread before all the copying has been finished. So I think it's easier now to keep the list alive in the search thread to avoid using mutex to access it.